### PR TITLE
port image_publisher on ROS2

### PR DESCRIPTION
**Pull request by yechun1**
* switch to use cmake 3.5
* change nodelet to classloader
* change ros::param to ros2 parameter APIs
* use ros2 code style

### CODE REVIEW

**Comment by yechun1** [comment]
Test result:

publisher png image
```
$ ros2 run image_publisher image_publisher /opt/ros/melodic/share/rviz/images/splash.png
[INFO] [ImagePublisherNode]: param_change_callback
[INFO] [ImagePublisherNode]: Reset filename as '/opt/ros/melodic/share/rviz/images/splash.png'
[INFO] [ImagePublisherNode]: File name for publishing image is : /opt/ros/melodic/share/rviz/images/splash.png
libpng warning: iCCP: known incorrect sRGB profile
[INFO] [ImagePublisherNode]: Flip horizontal image is : false
[INFO] [ImagePublisherNode]: Flip flip_vertical image is : false
```

publisher usbcamera frame
```
$ ros2 run image_publisher image_publisher 0
[INFO] [ImagePublisherNode]: param_change_callback
[INFO] [ImagePublisherNode]: Reset filename as '0'
[INFO] [ImagePublisherNode]: File name for publishing image is : 0
VIDEOIO ERROR: V4L2: setting property #1 is not supported
[INFO] [ImagePublisherNode]: Flip horizontal image is : false
[INFO] [ImagePublisherNode]: Flip flip_vertical image is : false
```

print topic
```
$ ros2 topic list
/camera_info
/image_raw
/parameter_events
```

print /camera_info
```
$ ros2 topic echo /camera_info
header:
  stamp:
    sec: 1543299331
    nanosec: 643252803
  frame_id: camera
height: 480
width: 640
distortion_model: plumb_bob
d: [0.0, 0.0, 0.0, 0.0, 0.0]
k: [1.0, 0.0, 320.0, 0.0, 1.0, 240.0, 0.0, 0.0, 1.0]
r: [1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0]
p: [1.0, 0.0, 320.0, 0.0, 0.0, 1.0, 240.0, 0.0, 0.0, 0.0, 1.0, 0.0]
binning_x: 0
binning_y: 0
roi:
  x_offset: 0
  y_offset: 0
  height: 0
  width: 0
  do_rectify: false
```

**Comment by yechun1** [comment]
@mjcarroll this PR is enabled image_publisher, pass local tests with image and usb camera. Would you please help to review the PR? thanks.

File: image_publisher/package.xml [hunk 1]
```diff
@@ -9,27 +9,28 @@
...
+  <buildtool_depend>ament_cmake</buildtool_depend>
+
+  <test_depend>ament_cmake_gtest</test_depend>
+  <test_depend>ament_lint_auto</test_depend>
+  <test_depend>ament_lint_common</test_depend>
 
   <build_depend>cv_bridge</build_depend>
-  <build_depend>dynamic_reconfigure</build_depend>
-  <build_depend>camera_info_manager</build_depend>
+  <!--build_depend>camera_info_manager</build_depend-->
```
Comment (mjcarroll) [inline image_publisher/package.xml]: camera_info_manager is available now (or once I get this last PR merged this morning)

**Comment by yechun1** [comment]
@mjcarroll when I porting camera_info_manager, I meet the same pointer API issue similar with image_transport.

for ros1 code:
camera_info_manager::CameraInfoManager c(nh_)

Not sure if we could modify apply **[Pointer api updates](https://github.com/ros-perception/image_common/pull/104)** in ros2 camera_info_mamanger, and pass "this" to 
camera_info_manager::CameraInfoManager c(this);

**Comment by yechun1** [comment]
@mjcarroll I have submitted PR to switch SharePtr to pointer on https://github.com/ros-perception/image_common/pull/106
and commit 21eb24b in this PR to enable camera_info_manager, would you please help to review both PRs. thanks.

**Review by mjcarroll — APPROVED** [review]

**Comment by mjcarroll** [comment]
Just saw that this wasn't building, reverting for now

**Comment by mjcarroll** [comment]
Ah, sorry, I somehow messed this up, I missed the `image_common` update.  I'll have to cut a new release over there.

**Pull request merged**